### PR TITLE
Add option to skip pull from cache

### LINKED ISSUE #46: Add flag to enable avoiding needing to pull the image on cache hit
Context:

Sometimes you may have a caching step ahead of time to ensure that the cache is available to pull in future steps. It is pointless to pull it if we know it exists as it just wastes time. Especially if your steps are ephemeral and wipe the cache for future steps anyway.

`--skipPullFromCache` or something along those lines would be nice.

I can work on this. This would save on average ~1-1:30 on each of my pipelines

### PR DESCRIPTION
**Pull request by samchungy**
Resolves #46

### CODE REVIEW

File: hooks/lib/gcr-registry-provider.bash [hunk 1]
```diff
@@ -18,3 +18,8 @@ get_registry_url() {
   fi
   echo "${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_REGISTRY_HOSTNAME}/${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_GCP_PROJECT}/${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_ECR_NAME:-"$(get_default_image_name)"}"
 }
+
+image_exists() {
+  # TODO - implement check for cache in GCR
```
Comment (samchungy) [inline hooks/lib/gcr-registry-provider.bash:23]: I don't intend to add this support for GCR, but if someone needs this, they can raise a PR.

File: tests/ecr-registry-provider.bats [hunk 2]
```diff
@@ -184,6 +184,45 @@ pre_command_hook="$PWD/hooks/pre-command"
   assert_output --partial "pushed deadbeef"
   assert_output --partial "pushed latest"
 
+  unstub aws
+  unstub docker
+  unstub sha1sum
+}
+
+@test "ECR: Calls list-images to check existance of cache" {
```
Comment (72636c) [inline tests/ecr-registry-provider.bats]:
```suggestion
@test "ECR: Calls list-images to check existence of cache" {
```

File: tests/ecr-registry-provider.bats [hunk 3]
```diff
@@ -184,6 +184,45 @@ pre_command_hook="$PWD/hooks/pre-command"
...
+    "ecr list-images --repository-name ${expected_repository_name} --query imageIds[?imageTag==\'deadbee\'].imageTag --output text : echo 'deadbee'"
+
+  stub docker \
+    "login --username AWS --password-stdin 1234567891012.dkr.ecr.ap-southeast-1.amazonaws.com : echo logging in to docker"
+
+  stub sha1sum \
+    "Dockerfile : echo 'sha1sum(Dockerfile)'" \
+    ": echo sha1sum" \
+    ": echo sha1sum" \
+    ": echo deadbeef"
```
Comment (72636c) [inline tests/ecr-registry-provider.bats]: Is it intentional that one is 💀🐝 and the other 💀🐄?

**Review by 72636c — APPROVED** [review]
🚀

File: tests/ecr-registry-provider.bats (hunk 3 above)
Comment (samchungy) [reply tests/ecr-registry-provider.bats]: copy pasta'd from another before 😅

Comment (samchungy) [reply tests/ecr-registry-provider.bats]: I'll re-name all. Seems like a fat finger

**Pull request merged**